### PR TITLE
sem: init at 0.28.1

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

Adding the semaphore cli 2.0. 
Interact and control your CI/CD from the terminal and operate on running jobs.

Source: https://github.com/semaphoreci/cli.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Signed-off-by: liberatys <liberatys@hey.com>

### CODE REVIEW

File: pkgs/development/tools/sem/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 = "sha256-Ztkrda84zogB18wcAL+9wzz5/zj2XafOd/UgdBuipj0=";
+  };
+
+  vendorSha256 = "sha256-fr88hZ8xxAeneRjUUDG/SrLu1hTU2C6iQltKgbjN+uU=";
+  subPackages = [ "." ];
+
+  ldflags = [ "-X main.version=${version}" "-X main.buildSource=nix" ];
+
+  postInstall = ''
+    mv $out/bin/cli $out/bin/sem
```
[the remark this answers is not in the document]
Comment (necrophcodr) [reply pkgs/development/tools/sem/default.nix:20]: @Liberatys yes, you may want to use `install -m755 infile outfile` instead.

**Comment by onny** [comment]
Please rebase with master

**Comment by onny** [comment]
Please rebase on top of master

**Comment by ryantm** [comment]
I rebased this, bumped the version, and tested basic operations of it myself. I think this is good to go and will merge when eval completes.

**Pull request merged**